CPE/fetch-and-update.py

```diff
@@ -103,38 +109,23 @@
 #     zf.write(BASEDIR+'/data/cpes.json', arcname='cpes.json')
 
 print("[+] Building diff file from latest CPE references")
-with open(BASEDIR+'/data/cpes-base.json', "r") as jfo:
-    cpes_oldies = json.loads(jfo.read())['cpes']
-    cpes_diffs = {}
+with open(BASEDIR + "/data/cpes-base.json") as jfo:
+    cpes_oldies = json.loads(jfo.read())["cpes"]
+    new_cpes = defaultdict(lambda: defaultdict(dict))
 
     # Loop over new vendors list
-    for n_vendor in cpes.keys():
-        for n_product in cpes[n_vendor].keys():
-            for n_cpe in cpes[n_vendor][n_product].keys():
+    for vendor, products in cpes.items():
+        for product, cpes in products.items():
+            for cpe, title in cpes.items():
                 try:
-                    cpes_oldies[n_vendor][n_product][n_cpe]
-                    is_new = False
-                except KeyError:  # Fuck les jaloux
-                    is_new = True
-
-                if is_new:
-                    cpes_diffs = {
-                        **cpes_diffs,
-                        **{
-                            n_vendor: {
-                                n_product: {
-                                    n_cpe: cpes[n_vendor][n_product][n_cpe]
-                                }
-                            }
-                        }
-                    }
-                    counters_diff.update({'cpes': counters_diff['cpes']+1})
+                    cpes_oldies[vendor][product][cpe]
+                except KeyError:
+                    new_cpes[vendor][product][cpe] = title
+                    counters_diff.update({"cpes": counters_diff["cpes"] + 1})
```

Comment:
I think it's better to use an if than raise an exception. The code will be small and easiest to understand.